src/drafty.js

```diff
@@ -2262,7 +2300,9 @@ function spansToTree(parent, text, start, end, spans) {
     // Add un-styled range before the styled span starts.
     if (start < span.start) {
       addNode(parent, {
-        text: text.substring(start, span.start)
+        text: graphemes.slice(start, span.start)
+          .map(segment => segment.segment)
+          .join("")
```

Comment:
Code style: please use ' (single quotes) for strings which require no localization.